### PR TITLE
Config: use hyprutils helper

### PR DESCRIPTION
**Pull request by fufexan**
#### Describe your PR, what does it fix/add?

Uses the helpers added in https://github.com/hyprwm/hyprutils/pull/8.
Follow-up PRs will also come to other hypr*apps near you.

Fixes https://github.com/hyprwm/Hyprland/issues/1040.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)

Tested all cases (any/all of the required variables missing, config missing completely).
Autogeneration also works.

#### Is it ready for merging, or does it need work?

Probably ready.

### CODE REVIEW

File: src/config/ConfigManager.cpp [hunk 1]
```diff
@@ -628,25 +629,49 @@ CConfigManager::CConfigManager() {
...
 std::string CConfigManager::getMainConfigPath() {
     if (!g_pCompositor->explicitConfigPath.empty())
         return g_pCompositor->explicitConfigPath;
 
-    return getConfigDir() + "/hypr/" + (ISDEBUG ? "hyprlandd.conf" : "hyprland.conf");
+    static const auto paths = Hyprutils::Path::findConfig(ISDEBUG ? "hyprlandd" : "hyprland");
+    if (paths.first.has_value()) {
+        return paths.first.value();
+    } else if (paths.second.has_value()) {
+        auto configPath = Hyprutils::Path::fullConfigPath(paths.second.value(), ISDEBUG ? "hyprlandd" : "hyprland");
```
Comment (vaxerski) [inline src/config/ConfigManager.cpp:662]: what if .second is empty

Comment (fufexan) [reply src/config/ConfigManager.cpp:662]: Can't be empty. It can only be `std::nullopt` in case there's no value.

Comment (vaxerski) [reply src/config/ConfigManager.cpp:662]: ok

**Review by vaxerski — APPROVED** [review]
cool but has to wait for a new hyprutils release

**Comment by vaxerski** [comment]
would you look at that https://github.com/hyprwm/hyprutils/releases/tag/v0.2.0

can you make this require 0.2

**Comment by fufexan** [comment]
Done.

**Comment by fufexan** [comment]
Ah autoformatting on `:wq`. Should I revert?

**Comment by vaxerski** [comment]
ye pls

**Review by vaxerski — APPROVED** [review]
サンキュー

**Pull request merged**